site/_filters/toc.js

```diff
@@ -40,8 +40,8 @@ const headingRe = /<(h[23])(|\s.*?)>(.*?)<\/\1>/gm;
 // Matches an ID attribute, returning the value in the 1st group.
 const idRe = /\bid="(.+?)"/;
 
-// Matches e.g., " #" so we can remove it.
-const trailingHashRe = /\s+#$/;
+// Matches e.g., " #" or "# " so we can remove it.
+const trailingHashRe = /(\s+#$)|(^#\s+)/g;
```

Comment:
Conceptually it's a little weird that the first half of the expression matches against the end of a string and the second part of the expression matches the start of a string. Reordering these might be a little more obvious to future readers.

```suggestion
const trailingHashRe = /(^#\s+)|(\s+#$)/g;
```

Additionally, the new pattern is no longer just a trailing hash. Should this variable be renamed? Perhaps something like "anchorLinkTextRe".

Reply:
Ack done.